internal/proxy/proxy.go

```diff
@@ -307,22 +240,50 @@ func (c *Client) Serve(ctx context.Context) error {
 	return <-exitCh
 }
 
+// MultiErr is a group of errors wrapped into one.
+type MultiErr []error
+
+// Error returns a single string representing one or more errors.
+func (m MultiErr) Error() string {
+	l := len(m)
+	if l == 1 {
+		return m[0].Error()
+	}
+	var errStr string
+	for i, err := range m {
+		if i == l-1 {
+			errStr += err.Error()
+			continue
+		}
+		errStr += err.Error() + ", "
+	}
```

Comment:
This seems potentially problematic since we're adding a lot of strings together and will have exponential runtime growth. Can we use `strings.Join()` instead?